frontend/src/metabase/query_builder/tests/common.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  createMockDatabase,
+  createMockField,
+  createMockTable,
+} from "metabase-types/api/mocks";
+import AtomicQuery from "metabase-lib/queries/AtomicQuery";
+import Field from "metabase-lib/metadata/Field";
+import { TYPE } from "metabase-lib/types/constants";
+import Table from "metabase-lib/metadata/Table";
+import Database from "metabase-lib/metadata/Database";
+import Metadata from "metabase-lib/metadata/Metadata";
+
+type Constructor<T> = new (...args: any[]) => T;
+
+// For some reason copying fields logic in Base class doesn't work in Storybook. Created instance has default values for all attributes.
+// So, we have to manually copy values into newly created instance.
+const createNewInstance = <T>(
+  Class: Constructor<T>,
+  object: Record<string, unknown>,
+): T => {
+  const instance = new Class(object);
+
+  for (const property in object) {
+    (instance as any)[property] = object[property];
+  }
+
+  return instance;
+};
```

Comment:
I believe we need to figure out a way to avoid this, I'll try a few things locally

Reply:
let me know if you came up with anything. I've tried, and it seems like the constructor here is doing its work properly: https://github.com/metabase/metabase/blob/65ab08c4266a3a359dec776b0bfc8669bb9b8457/frontend/src/metabase-lib/metadata/Base.ts#L9-L11

I logged that `this[property]`, and they seem to have values, but if I log `this` and observe its value at runtime, some of the values will be gone. It seems as if they're overridden somehow 🤔

Reply:
Yeah, something weird is going on. IMO if we can't find a cleaner way to fix this, I'd consider removing `ExpressionWidget` storybook file :( We have plenty of magic in the codebase already, and I'm worried about adding more. @WiNloSt @ranquild WDYT?

Reply:
I think it should be possible to use the helper from https://github.com/metabase/metabase/pull/28836 in storybook. Hopefully no metabase-lib changes would be required. @deniskaber could you take a look?

Reply:
I think the benefit of having a storybook to play with, especially when modifying the component, outweighs the quirkiness we have here. I'm inclined to leave the hack here.

Reply:
Rendering components dealing with `metabase-lib` stuff can be complicated. Maybe we could eventually refactor `ExpressionWidget` to be independent from `metabase-lib`; adding it to Storybook would be straightforward in this case

Reply:
@kulyk btw WDYT about that?

Reply:
@ranquild sorry, we pulled a few different threads here, so I'm not sure I understand what you are referring to

> I think it should be possible to use the helper from #28836 in storybook. Hopefully, no metabase-lib changes would be required.

That might work; it'd be great if we could give it a shot

> I think the benefit of having a storybook to play with, especially when modifying the component, outweighs the quirkiness we have here. I'm inclined to leave the hack here.

If we keep these hacks/workarounds, can we be sure that components in Storybook would behave the same way as in the core app? I personally find a solid RTL test suite much more helpful when modifying a component; the tests would immediately highlight changes in behavior.

I'm probably biased because I don't use Storybook often enough here. So if the rest of the team finds it useful — let's keep it. I'd only reconsider where that code should live (probably something more top-level than the QB directory)

Reply:
I hope that no hacks would be needed if the helper is used. If hacks are still required, let's discuss this again. @kulyk @deniskaber @WiNloSt

Reply:
> I'm probably biased because I don't use Storybook often enough here. So if the rest of the team finds it useful — let's keep it. I'd only reconsider where that code should live (probably something more top-level than the QB directory)

I used to use Storybook often, but the way some components in metabase are structured makes it hard to render in storybook or even impossible for some components. Ideally, we should be able to render any pure UI in storybook, but we seem to use our entity loaders and redux here and there.

Usually, I would run Storybook to develop any pure UI or styling change, but I have never been able to do that here 😓 

re:
> I hope that no hacks would be needed if the helper is used. If hacks are still required, let's discuss this again.

And 

> If we keep these hacks/workarounds, can we be sure that components in Storybook would behave the same way as in the core app?

I also agree with you guys, so let's see if we could get rid of this workaround.